internal/state/execution.go

```diff
@@ -152,6 +152,23 @@ func (blockExec *BlockExecutor) CreateProposalBlock(
 	return state.MakeBlock(height, modifiedTxs, commit, evidence, proposerAddr)
 }
 
+func (blockExec *BlockExecutor) ProcessProposal(
```

Comment:
This should take a context instead of using `context.Background`. There's one in scope when this is called.

Reply:
OK, I added a new param and passed it from the caller. Please double-check how I addressed this.
IMPORTANT: Please look at line 122 above (bumped into it by chance while trying to address this comment); do we need to fix that instance of `context.Background()` as well?